### PR TITLE
docs: update the angular roadmap for v19

### PR DESCRIPTION
**Pull request by mgechev**

### CODE REVIEW

File: adev/src/content/reference/roadmap.md [hunk 1]
```diff
@@ -19,22 +19,24 @@ Start developing with the latest Angular features from our roadmap. This list re
 
 ### Available to experiment with
 
-* [Explore Angular Signals](guide/signals)
-* [Event replay with SSR](https://angular.dev/api/platform-browser/withEventReplay)
+* [Incremental hydration](https://angular.dev/guide/experimental/incremental-hydration)
 * [Zoneless change detection](https://angular.dev/guide/experimental/zoneless)
 * [Hydration support for i18n blocks](https://angular.dev/api/platform-browser/withI18nSupport)
+* Resource API
+* Effect API
```
Comment (pkozlowski-opensource) [inline adev/src/content/reference/roadmap.md:26]: Did you mean `afterRenderEffect`? Or is `afterRenderEffect` missing from this list?

File: adev/src/content/reference/roadmap.md [hunk 2]
```diff
@@ -19,22 +19,24 @@ Start developing with the latest Angular features from our roadmap. This list re
...
 * [Deferrable views](https://angular.dev/guide/defer)
 * [Built-in control flow](https://angular.dev/guide/templates/control-flow)
-* [Migrate your Angular Material to MDC](https://material.angular.io/guide/mdc-migration)
-* [Angular Material 3](https://material.angular.io/guide/theming)
-* [Migrate to Standalone APIs](reference/migrations/standalone)
-* [Improve image performance with NgOptimizedImage](guide/image-optimization)
-* [Try out Inject](/tutorials/learn-angular/20-inject-based-di)
-* [New CDK directives](https://material.angular.io/cdk/categories)
+* [Local variable declaration](https://angular.dev/guide/templates/variables)
+* [Signal inputs](https://angular.dev/guide/signals/inputs)
```
Comment (pkozlowski-opensource) [inline adev/src/content/reference/roadmap.md:37]: Should we mention `model` as well?

File: adev/src/content/reference/roadmap.md [hunk 3]
```diff
@@ -103,34 +112,50 @@ Start developing with the latest Angular features from our roadmap. This list re
...
+  <docs-card title="Support two-dimensional drag-and-drop" link="Completed in Q2 2024" href="https://github.com/angular/components/issues/13372">
+  As part of this project, we implemented mixed orientation support for the Angular CDK drag and drop. This is one of the repository's most highly requested features.
+  </docs-card>
+  <docs-card title="Event replay with SSR and prerendering" link="Completed in Q4 2024" href="https://angular.dev/api/platform-browser/withEventReplay">
+  In v18 we introduced an event replay functionality when using server-side rendering or prerendering. For this feature we depend on the event dispatch primitive (previously known as jsaction) that is running on Google.com.
+
+  In Angular v19 we graduated event replay to stable and enabled it by default for all new projects.
+  </docs-card>
+  <docs-card title="Integrate Angular Language Service with Schematics" link="Completed in Q4 2024" href="">
+  To make it easier for developers to use modern Angular APIs, we enabled integration between the Angular language service and schematics which allows you to refactor your app with a single click.
```
Comment (msmallest) [inline adev/src/content/reference/roadmap.md:151]: Is this point about something like the individual signal input functionality? I think that would be a good example to give here.

File: adev/src/content/reference/roadmap.md [hunk 4]
```diff
@@ -44,15 +46,23 @@ Start developing with the latest Angular features from our roadmap. This list re
...
   </docs-card>
-  <docs-card title="Local template variables" href="https://github.com/angular/angular/issues/15280">
-  Local template variables is one of the most upvoted features in the Angular issue tracker. In Q2 2024 we started initial design and prototyping. Expect updates later in 2024.
+  <docs-card title="Signal integrations" href="">
+  We're working towards improving the integration of fundamental Angular packages, such as forms, HTTP, and router, with Signals. As part of this project, we'll seek opportunities to introduce convenient signal-based APIs or wrappers to improve the holistic developer experience.
+  </docs-card>
+  <docs-card title="Signal debugging in Angular DevTools" href="">
+  With the evolution of Signals in Angular, we are working on a better tooling for debugging them. High on the priority list is a UI for inspecting and debugging Signal-based components.
```
Comment (eneajaho) [inline adev/src/content/reference/roadmap.md:60]:
```suggestion
  With the evolution of Signals in Angular, we are working on a better tooling for debugging them. High on the priority list is a UI for inspecting and debugging signal usages in components.
```

Signal-based components would connect the current impl of zone-based components to signal: true components which is not the case yet.

File: adev/src/content/reference/roadmap.md [hunk 5]
```diff
@@ -71,30 +81,29 @@ Start developing with the latest Angular features from our roadmap. This list re
 
 <docs-card-container>
   <docs-card title="Modernize unit testing tooling with ng test" href="">
-  In v12, we revisited the Angular end-to-end testing experience by replacing Protractor with modern alternatives such as Cypress, Nightwatch, and Webdriver.io. Next, we'd like to tackle `ng test` to modernize Angular's unit testing experience. In Q2 2023, we introduced experimental [Jest](https://jestjs.io/) support and [announced](https://blog.angular.dev/moving-angular-cli-to-jest-and-web-test-runner-ef85ef69ceca) the transition from Karma to the [Web Test Runner](https://modern-web.dev/docs/test-runner/overview/).
+  In v12, we revisited the Angular end-to-end testing experience by replacing Protractor with modern alternatives such as Cypress, Nightwatch, and Webdriver.io. Next, we'd like to tackle `ng test` to modernize Angular's unit testing experience.
 
-  Later this year we'll continue making progress towards introducing Web Test Runner as the replacement of Karma.
+  We're currently evaluating Web Test Runner, Vitest, and Jest as candidates for a new test runner for Angular projects while preserving Jasmine as assertion library to not break existing tests.
```
Comment (yjaaidi) [inline adev/src/content/reference/roadmap.md:86]: Super excited to see Vitest in this list. Thank you!!

File: adev/src/content/reference/roadmap.md [hunk 6]
```diff
@@ -103,34 +112,50 @@ Start developing with the latest Angular features from our roadmap. This list re
...
+  In Angular v19 we graduated event replay to stable and enabled it by default for all new projects.
+  </docs-card>
+  <docs-card title="Integrate Angular Language Service with Schematics" link="Completed in Q4 2024" href="">
+  To make it easier for developers to use modern Angular APIs, we enabled integration between the Angular language service and schematics which allows you to refactor your app with a single click.
+  </docs-card>
+  <docs-card title="Streamline standalone imports with Language Service" link="Completed in Q4 2024" href="">
+  As part of this initiative, the language service automatically imports components and pipes in standalone and NgModule-based apps. Additionally, to enable smaller app bundles, we'll work on allowing the language service to propose the automatic removal of unused imports.
+  </docs-card>
+  <docs-card title="Local template variables" link="Completed in Q3 2024" href="https://github.com/angular/angular/issues/15280">
+  Local template variables is one of the most upvoted features in the Angular issue tracker. In Q2 2024 we started initial design and prototyping. Expect updates later in 2024.
```
[the remark this answers is not in the document]
Comment (mgechev) [reply adev/src/content/reference/roadmap.md:157]: Nothing more to expect about local template variables.

File: adev/src/content/reference/roadmap.md (hunk 3 above)
Comment (mgechev) [reply adev/src/content/reference/roadmap.md:151]: We're explaining this in the upcoming v19 blog post. That's just about the integration between the two.

File: adev/src/content/reference/roadmap.md (hunk 4 above)
Comment (mgechev) [reply adev/src/content/reference/roadmap.md:60]: Unsure if it'll be `signal: true`, but yeah. It should be debugging of signals overall, not signal-based components.

File: adev/src/content/reference/roadmap.md [hunk 7]
```diff
@@ -19,22 +19,25 @@ Start developing with the latest Angular features from our roadmap. This list re
...
 
-* [Explore Angular Signals](guide/signals)
-* [Event replay with SSR](https://angular.dev/api/platform-browser/withEventReplay)
+* [Incremental hydration](https://angular.dev/guide/experimental/incremental-hydration)
 * [Zoneless change detection](https://angular.dev/guide/experimental/zoneless)
 * [Hydration support for i18n blocks](https://angular.dev/api/platform-browser/withI18nSupport)
+* Resource API
+* Effect API
+* Linked Signal API
+* Route-level render mode
```
Comment (AndrewKushnir) [inline adev/src/content/reference/roadmap.md:28]:
```suggestion
* [Route-level render mode](https://next.angular.dev/guide/hybrid-rendering)
```

File: adev/src/content/reference/roadmap.md (hunk 1 above)
Comment (AndrewKushnir) [reply adev/src/content/reference/roadmap.md:26]: Should we also link those items to corresponding pages?

File: adev/src/content/reference/roadmap.md [hunk 8]
```diff
@@ -44,15 +47,23 @@ Start developing with the latest Angular features from our roadmap. This list re
...
+  <docs-card title="Signal integrations" href="">
+  We're working towards improving the integration of fundamental Angular packages, such as forms, HTTP, and router, with Signals. As part of this project, we'll seek opportunities to introduce convenient signal-based APIs or wrappers to improve the holistic developer experience.
+  </docs-card>
+  <docs-card title="Signal debugging in Angular DevTools" href="">
+  With the evolution of Signals in Angular, we are working on a better tooling for debugging them. High on the priority list is a UI for inspecting and debugging signals.
+  </docs-card>
+  <docs-card title="Improve HMR (Hot Module Reload)" href="https://github.com/angular/angular/issues/39367#issuecomment-1439537306">
+  We're working towards faster edit/refresh cycle by enabling hot module replacement.
+  
+  In Angular v19 we shipped initial support for CSS and HTML HMR. We'll continue collecting feedback to make sure we're addressing developers' needs before we mark this project as complete.
```
Comment (AndrewKushnir) [inline adev/src/content/reference/roadmap.md:66]:
```suggestion
  In Angular v19 we shipped initial support for CSS and template HMR. We'll continue collecting feedback to make sure we're addressing developers' needs before we mark this project as complete.
```

File: adev/src/content/reference/roadmap.md [hunk 9]
```diff
@@ -71,30 +82,29 @@ Start developing with the latest Angular features from our roadmap. This list re
...
 
-  Later this year we'll continue making progress towards introducing Web Test Runner as the replacement of Karma.
+  We're currently evaluating Web Test Runner, Vitest, and Jest as candidates for a new test runner for Angular projects while preserving Jasmine as assertion library to not break existing tests.
   </docs-card>
-  <docs-card title="Streamline standalone imports with Language Service" href="">
-  As part of this initiative, the language service automatically imports components and pipes in standalone and NgModule-based apps. Additionally, to enable smaller app bundles, we'll work on allowing the language service to propose the automatic removal of unused imports.
+  <docs-card title="Evaluating Nitro support in the Angular CLI" href="https://nitro.unjs.io/">
+  We're excited about the set of features that Nitro offers such as portability, minimal design, and file-based routing. In 2025 we'll evaluate how it fits in the Angular server-side rendering model.
```
Comment (AndrewKushnir) [inline adev/src/content/reference/roadmap.md:90]:
```suggestion
  We're excited about the set of features that Nitro offers such as more deployment options, improved compatibility of server-side rendering with different runtimes and file-based routing. In 2025 we'll evaluate how it fits in the Angular server-side rendering model.
```

File: adev/src/content/reference/roadmap.md [hunk 10]
```diff
@@ -71,30 +82,29 @@ Start developing with the latest Angular features from our roadmap. This list re
...
-  </docs-card>
-  <docs-card title="Event replay with SSR and prerendering" href="https://angular.dev/api/platform-browser/withEventReplay">
-  In v18 we introduced an event replay functionality when using server-side rendering or prerendering. For this feature we depend on the event dispatch primitive (previously known as jsaction) that is running on Google.com.
-
-  Over the next couple of months we'll be collecting feedback from the community for this feature and work towards graduating it to stable.
+  <docs-card title="Enable incremental hydration" href="">
+  In v17 we graduated hydration from developer preview and we've been consistently observing 40-50% improvements in LCP. Since then we started prototyping incremental hydration and shared a demo on stage at ng-conf.
+  
+  In v19 we shipped an experimental version of incremental hydration that you try using deferrable views. Give it a try and <a href="https://github.com/angular/angular/issues">share your feedback</a> with us!
```
Comment (AndrewKushnir) [inline adev/src/content/reference/roadmap.md:102]:
```suggestion
  In v19 we shipped the incremental hydration in developer preview mode, powered by `@defer` blocks. Give it a try and <a href="https://github.com/angular/angular/issues">share your feedback</a> with us!
```

File: adev/src/content/reference/roadmap.md [hunk 11]
```diff
@@ -103,34 +113,50 @@ Start developing with the latest Angular features from our roadmap. This list re
 This section represents explorations and prototyping of potential future projects. A reasonable outcome is to decide that our current solutions are the best options. Other projects may result in RFCs, graduating to in-progress projects, or being deprioritized as the web continues to innovate along with our framework.
 
 <docs-card-container>
-  <docs-card title="Signal debugging in Angular DevTools" href="">
-  With the evolution of Signals in Angular, we'll be also working on a better tooling for debugging them. High on the priority list is a UI for inspecting and debugging Signal-based components.
+  <docs-card title="Signal Forms" href="">
+  Analyze existing feedback about Angular forms and design a solution which addresses developers' requirements and uses Signals for management of reactive state.
```
Comment (AndrewKushnir) [inline adev/src/content/reference/roadmap.md:117]:
```suggestion
  We plan to analyze existing feedback about Angular forms and design a solution which addresses developers' requirements and uses Signals for management of reactive state.
```

File: adev/src/content/reference/roadmap.md [hunk 12]
```diff
@@ -103,34 +113,50 @@ Start developing with the latest Angular features from our roadmap. This list re
...
   </docs-card>
   <docs-card title="Investigation for authoring format improvements" href="">
   Based on our developer surveys' results we saw there are opportunities for improving the ergonomics of the component authoring format. The first step of the process will be to gather requirements and understand the problem space in advanced to an RFC. We'll share updates as we make progress. High priority in the future work will be backward compatibility and interoperability.
   </docs-card>
-  <docs-card title="Support two-dimensional drag-and-drop" href="https://github.com/angular/components/issues/13372">
-  As part of this project, we'd like to implement mixed orientation support for the Angular CDK drag and drop. This is one of the repository's most highly requested features.
+  <docs-card title="Improve TestBed" href="">
+  Based on feedback over the years and the recent updates in Angular's runtime, we'll evaluate TestBed to improve developer experience and reduce boilerplate when developing unit tests.
```
Comment (AndrewKushnir) [inline adev/src/content/reference/roadmap.md:131]:
```suggestion
  Based on feedback over the years and the recent updates in Angular's runtime, we'll evaluate TestBed to identify opportunities to improve developer experience and reduce boilerplate when developing unit tests.
```

File: adev/src/content/reference/roadmap.md [hunk 13]
```diff
@@ -103,34 +113,50 @@ Start developing with the latest Angular features from our roadmap. This list re
...
+  In v18 we introduced an event replay functionality when using server-side rendering or prerendering. For this feature we depend on the event dispatch primitive (previously known as jsaction) that is running on Google.com.
+
+  In Angular v19 we graduated event replay to stable and enabled it by default for all new projects.
+  </docs-card>
+  <docs-card title="Integrate Angular Language Service with Schematics" link="Completed in Q4 2024" href="">
+  To make it easier for developers to use modern Angular APIs, we enabled integration between the Angular language service and schematics which allows you to refactor your app with a single click.
+  </docs-card>
+  <docs-card title="Streamline standalone imports with Language Service" link="Completed in Q4 2024" href="">
+  As part of this initiative, the language service automatically imports components and pipes in standalone and NgModule-based apps. Additionally, to enable smaller app bundles, we'll work on allowing the language service to propose the automatic removal of unused imports.
```
Comment (AndrewKushnir) [inline adev/src/content/reference/roadmap.md:155]:
```suggestion
  As part of this initiative, the language service automatically imports components and pipes in standalone and NgModule-based apps. Additionally, we've added a template diagnostic to highlight unused imports in standalone components, which should help make application bundles smaller.
```

File: adev/src/content/reference/roadmap.md [hunk 14]
```diff
@@ -103,34 +113,50 @@ Start developing with the latest Angular features from our roadmap. This list re
...
+  </docs-card>
+  <docs-card title="Local template variables" link="Completed in Q3 2024" href="https://github.com/angular/angular/issues/15280">
+  Local template variables is one of the most upvoted features in the Angular issue tracker. In Q2 2024 we started initial design and prototyping. Expect updates later in 2024.
```
Comment (AndrewKushnir) [inline adev/src/content/reference/roadmap.md:158]:
```suggestion
  We've released the support for local template variables in Angular, see [`@let` docs](https://angular.dev/api/core/@let) for additional information.
```

**Review by AndrewKushnir — COMMENTED** [review]
LGTM, just proposed a few minor updates.

File: adev/src/content/reference/roadmap.md (hunk 7 above)
Comment (JeanMeche) [reply adev/src/content/reference/roadmap.md:28]: note: It's should be a relative route, as the guide update is targeting major

File: adev/src/content/reference/roadmap.md [hunk 15]
```diff
@@ -19,22 +19,25 @@ Start developing with the latest Angular features from our roadmap. This list re
 
 ### Available to experiment with
 
-* [Explore Angular Signals](guide/signals)
-* [Event replay with SSR](https://angular.dev/api/platform-browser/withEventReplay)
+* [Incremental hydration](https://angular.dev/guide/experimental/incremental-hydration)
```
Comment (JeanMeche) [inline adev/src/content/reference/roadmap.md:22]: Let's use relative urls, so when on next, we stay on the same version on the doc. 
```suggestion
* [Incremental hydration](/guide/experimental/incremental-hydration)
```

File: adev/src/content/reference/roadmap.md (hunk 8 above)
Comment (eneajaho) [reply adev/src/content/reference/roadmap.md:66]: Should we mention that it works on external templates only?

Comment (AndrewKushnir) [reply adev/src/content/reference/roadmap.md:66]: @clydin do we support an experimental HMR for inline templates in v19?

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 257fedc25a2693d7d8e1c2e020b74185736e3d26.

The changes were merged into the following branches: main, 19.0.x

**Pull request closed without merging**